src/MQTTClient.cpp

```diff
@@ -285,7 +312,19 @@ bool MQTTClient::connect(const char clientId[], const char username[], const cha
 
   // connect to host
   if (!skip) {
-    int ret = this->netClient->connect(this->hostname, (uint16_t)this->port);
+	int ret = 0;
+	if(this->hostname != nullptr){
+	  ret = this->netClient->connect(this->hostname, (uint16_t)this->port);
+	}else{
+	  uint32_t thisIP = 0;
+	  thisIP = ipaddress[0];
+	  thisIP |= uint32_t(ipaddress[1] << 8);
+	  thisIP |= uint32_t(ipaddress[2] << 16);
+	  thisIP |= uint32_t(ipaddress[3] << 24);
+	  
+	  ret = this->netClient->connect(thisIP, (uint16_t)this->port);
```

Comment:
Isn't the standard way to pass the IPAddress directly as the first parameter?

Reply:
Yeah, not sure what the error was I was getting initially that made me think that. It seems to be working with IPAddress.